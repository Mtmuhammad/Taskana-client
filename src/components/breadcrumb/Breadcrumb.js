// Breadcrumb component to indicate the current page's location
import React from 'react'

const Breadcrumb = ({ page }) => {
  return (
    <div data-testid="breadcrumb" className="page-breadcrumb border-bottom">
      <div className="row">
        <div
          className="
                col-lg-3 col-md-4 col-xs-12
                justify-content-start
                d-flex
                align-items-center
              "
        >
          <h5
            data-testid="breadcrumb-title"
            className="h3 font-weight-medium text-uppercase mb-0"
          >
            {page}
          </h5>
        </div>
        <div
          className="
                col-lg-9 col-md-8 col-xs-12
                d-flex
                justify-content-start justify-content-md-end
                align-self-center
              "
        >
          <nav data-testid="pages" aria-label="breadcrumb" className="mt-2">
            <ol className="breadcrumb mb-0 p-0">
              <li className="h4 breadcrumb-item">Home</li>
              <li className="h4 breadcrumb-item">{page}</li>
            </ol>
          </nav>
        </div>
      </div>
    </div>
  );
};

export default Breadcrumb;
