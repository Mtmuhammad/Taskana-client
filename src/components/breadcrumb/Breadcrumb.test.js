import React from "react";
import { render, screen } from "@testing-library/react";
import Breadcrumb from "./Breadcrumb";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<Breadcrumb page="Dashboard" />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<Breadcrumb page="Dashboard" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("breadcrumb", () => {
  test("should render breadcrumb", () => {
    render(<Breadcrumb page="Dashboard" />);
    const breadcrumb = screen.getByTestId("breadcrumb");
    expect(breadcrumb).toBeInTheDocument();
    expect(breadcrumb).toHaveClass("page-breadcrumb");
  });
  test("should render breadcrumb title", () => {
    render(<Breadcrumb page="Dashboard" />);
    const title = screen.getByTestId("breadcrumb-title");
    expect(title).toBeInTheDocument();
  });
  test("should render breadcrumb pages", () => {
    render(<Breadcrumb page="Dashboard" />);
    const title = screen.getByTestId("pages");
    expect(title).toBeInTheDocument();
  });
});
