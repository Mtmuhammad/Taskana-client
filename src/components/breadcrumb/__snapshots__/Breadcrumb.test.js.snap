// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-breadcrumb border-bottom"
    data-testid="breadcrumb"
  >
    <div
      class="row"
    >
      <div
        class="
 col-lg-3 col-md-4 col-xs-12
 justify-content-start
 d-flex
 align-items-center
 "
      >
        <h5
          class="h3 font-weight-medium text-uppercase mb-0"
          data-testid="breadcrumb-title"
        >
          Dashboard
        </h5>
      </div>
      <div
        class="
 col-lg-9 col-md-8 col-xs-12
 d-flex
 justify-content-start justify-content-md-end
 align-self-center
 "
      >
        <nav
          aria-label="breadcrumb"
          class="mt-2"
          data-testid="pages"
        >
          <ol
            class="breadcrumb mb-0 p-0"
          >
            <li
              class="h4 breadcrumb-item"
            >
              Home
            </li>
            <li
              class="h4 breadcrumb-item"
            >
              Dashboard
            </li>
          </ol>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
